README.md

```diff
@@ -1,2 +1,331 @@
+
+Farcaster's Identity System is composed of two smart contracts - an **Account contract**, which creates new accounts, and a **Namespace contract** which issues usernames that can be used with accounts.
+
+## 3.1 Account Contract
+
+An account is created by calling `register` on the contract using a user-controlled Ethereum address. A new account number is issued and linked to this address. Accounts can be transferred between addresses, though the contract ensures that an address owns no more than one account at a time. 
+
+Account numbers start at 0 and are incremented by 1 every time a new account is registered, which is a gas efficient way to ensure unique account numbers. The account number is represented internally as a `uint256` counter and can be incremented up to ~ 10^77 which is practically infinite for our needs. 
+
+## 3.2 Namespace Contract
+
+A Farcaster Name like `@alice` can be minted from the Namespace contract and used on the protocol. 
+
+When minting a name, the contract checks that the name is unique and contains at most 16 alphanumeric characters or dashes `^[a-zA-Z0-9-]{1,16}$`. The single dash (-) username is reserved for a system account and will not be claimable. The mint happens over a two-phase commit reveal to prevent frontrunning registrations.
+
+Farcaster Names are ERC-721 tokens that are fully composable with the NFT ecosystem. While users can use ENS names with the protocol, Farcaster Names have some properties that make them more practical. Names are cheaper to mint and can be [recovered](#recovery) if the address holding them is lost. They are also less vulnerable to [homoglyph attacks](https://en.wikipedia.org/wiki/IDN_homograph_attack) and are easier to display because of the restricted length and character set. 
+
+A yearly fee of 0.01 ETH is charged for owning a name, and this must be renewed yearly on Farcaster Day (August 1st). Users who do not renew their usernames by this day have a 30-day grace period after which their ownership over the name expires. During mint, the fee is pro-rated for the year ending August 1st. Fees are collected by the Farcaster Treasury and are used to support protocol development.
+
+Names can be minted freely, but are subject to two policies enforced by governance: 
+
+1. During the first year, a "prior ownership" claim can be made if the user already owns the name on at least two major social media platforms, and the current owner does not own them on any platform. 
+
+2. During later years, community moderators can propose an "impersonation claim" to governance vote if someone is clearly impersonating or misleading users. 
```

Comment:
Who are community moderators? Not defined until now, so confusing. Maybe just "governance participating users" ?